h2o-py/tests/testdir_jira/pyunit_pubdev_5790.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import print_function
+import sys
+
+sys.path.insert(1, "../../")
+import h2o
+import csv
+import random
+import os
+import tempfile
+from tests import pyunit_utils
+
+#
+def csv_header_test():
+    components = []
+    for idx in range(704):
+        components.append('Field{idx}')
+
+    # 1000 columns
+    keys = list(set('_'.join(random.sample(components, 5)) for _ in range(1000)))
+
+    fieldnames = keys[:idx]
+    fd, temp_filename = tempfile.mkstemp()
+    try:
+        with open(temp_filename, 'w') as file_:
+            writer = csv.DictWriter(file_, fieldnames=fieldnames)
+            writer.writeheader()
+
+            # write 100 lines, resulting chunk size should be very small
```

Comment:
this depends on the system where h2o is running

if you need to control chunk size, you can specify it the parse api
or move the test to java and you have a total control over the chunking